react/internal/http-switch-link.js

```diff
@@ -7,42 +7,51 @@
  *
  */
 import {ApolloLink} from 'apollo-link'
-import { prop, map, contains, isEmpty, isNil } from 'ramda'
-import { parse, stringify } from 'querystring'
-import { print } from 'graphql'
+import {parse, format} from 'url'
+import {print} from 'graphql'
+
+const isEmpty = (val) => (val === undefined || val === "" || val === {} || val === [])
+const isNil = (val) => (val === undefined || val === null)
 
 function removeUnusedFields(obj) {
   Object.keys(obj).forEach(key => (isNil(obj[key]) || isEmpty(obj[key])) && delete obj[key])
 
   return obj
 }
 
+const hasMutationField = (queryTree) => {
+  for (let query of queryTree.definitions) {
+    if (query.operation === 'mutation') {
+      return true
+    }
+  }
+  return false
+}
+
 export const createHttpSwitchLink = (uri) => {
-  const [baseUrl, querystring] = uri.split('?')
-  let queryObj = (querystring) ? parse(querystring) : {}
+  const parsedUri = parse(uri, true)
 
   return new ApolloLink((operation, forward) => {
-    let targetUri = uri
+    const targetUri = Object.assign({}, parsedUri)
     const {query, variables, operationName} = operation
     const {fetchOptions = {}, http: httpOptions = {}} = operation.getContext()
-    const method = contains('mutation',map(prop('operation'), prop('definitions')(query))) ?
-      'POST' :
-      'GET'
+    const method = hasMutationField(query) ? 'POST' : 'GET'
 
     if(method === 'GET') {
       fetchOptions['method'] = method
 
-      const fullQueryObj = removeUnusedFields({
-        ...queryObj,
+      targetUri.query = removeUnusedFields({
+        ...targetUri.query,
         'query': print(query).replace(/\s\s+/g, ' '),
         variables,
         operationName
       })
 
-      targetUri = baseUrl.concat(`?${stringify(fullQueryObj)}`)
+      // need to delete in order for format function to work parsing the query instead of search field
+      delete targetUri.search
     }
 
-    operation.setContext({uri: targetUri, fetchOptions})
+    operation.setContext({uri: format(targetUri), fetchOptions})
```

Comment:
It's best to pass the old context here together with this new information, as otherwise some data may be lost (`setContext` replaces the last context, as I found out when using it inside `graphql-server`, it doesn't just add the new data to it).

So in this case, something like:

```js
operation.setContext({
  ...operation.getContext(),
  uri: format(targetUri), fetchOptions
})
```